ava.lang.NoClassDefFoundError: org/w3c/dom/ElementTraversal

Runtime exception, when run example org.apache.jena.examples.ExampleAPI_01

java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.jena.rdf.model.impl.IO_Ctl.callByRefection(IO_Ctl.java:64)
	at org.apache.jena.rdf.model.impl.IO_Ctl.init(IO_Ctl.java:36)
	at org.apache.jena.util.FileManager.<clinit>(FileManager.java:86)
	at org.apache.jena.examples.ExampleAPI_01.main(ExampleAPI_01.java:31)
Caused by: java.lang.NoClassDefFoundError: org/w3c/dom/ElementTraversal
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.apache.xerces.parsers.AbstractDOMParser.startDocument(Unknown Source)
	at org.apache.xerces.impl.dtd.XMLDTDValidator.startDocument(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl.startEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.startDocumentParsing(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at sun.util.xml.PlatformXmlPropertiesProvider.getLoadingDoc(PlatformXmlPropertiesProvider.java:106)
	at sun.util.xml.PlatformXmlPropertiesProvider.load(PlatformXmlPropertiesProvider.java:78)
	at java.util.Properties$XmlSupport.load(Properties.java:1201)
	at java.util.Properties.loadFromXML(Properties.java:881)
	at org.apache.jena.util.Metadata.read(Metadata.java:75)
	at org.apache.jena.util.Metadata.addMetadata(Metadata.java:53)
	at org.apache.jena.util.Metadata.<init>(Metadata.java:47)
	at org.apache.jena.JenaRuntime.<clinit>(JenaRuntime.java:89)
	at org.apache.jena.riot.system.stream.JenaIOEnvironment.getGlobalConfigPath(JenaIOEnvironment.java:69)
	at org.apache.jena.riot.system.stream.JenaIOEnvironment.getLocationMapper(JenaIOEnvironment.java:45)
	at org.apache.jena.riot.system.stream.StreamManager.makeDefaultStreamManager(StreamManager.java:73)
	at org.apache.jena.riot.system.stream.StreamManager.<clinit>(StreamManager.java:85)
	at org.apache.jena.riot.adapters.AdapterFileManager.makeGlobal(AdapterFileManager.java:147)
	at org.apache.jena.riot.adapters.AdapterFileManager.get(AdapterFileManager.java:106)
	at org.apache.jena.riot.IO_Jena.wireIntoJena(IO_Jena.java:35)
	at org.apache.jena.riot.RIOT.init(RIOT.java:66)
	... 8 more
Caused by: java.lang.ClassNotFoundException: org.w3c.dom.ElementTraversal
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 45 more
Exception in thread "main" java.lang.NoClassDefFoundError: Could not initialize class org.apache.jena.JenaRuntime
	at org.apache.jena.util.LocationMapper.getGlobalConfigPath(LocationMapper.java:241)
	at org.apache.jena.util.LocationMapper.get(LocationMapper.java:69)
	at org.apache.jena.util.FileManager.makeGlobal(FileManager.java:129)
	at org.apache.jena.util.FileManager.get(FileManager.java:95)
	at org.apache.jena.examples.ExampleAPI_01.main(ExampleAPI_01.java:31)




Later, added following dependency in pom.xml to solve this issue.
<dependency> 
    <groupId>xml-apis</groupId>
    <artifactId>xml-apis</artifactId>
    <version>1.4.01</version>
</dependency>

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,124 @@
-target
-tmp
+### Java template
+*.class
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.ear
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff:
+.idea/workspace.xml
+.idea/tasks.xml
+.idea/dictionaries
+.idea/vcs.xml
+.idea/jsLibraryMappings.xml
+
+# Sensitive or high-churn files:
+.idea/dataSources.ids
+.idea/dataSources.xml
+.idea/dataSources.local.xml
+.idea/sqlDataSources.xml
+.idea/dynamic.xml
+.idea/uiDesigner.xml
+
+# Gradle:
+.idea/gradle.xml
+.idea/libraries
+
+# Mongo Explorer plugin:
+.idea/mongoSettings.xml
+
+## File-based project format:
+*.iws
+
+## Plugin-specific files:
+
+# IntelliJ
+/out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
+### Eclipse template
+
+.metadata
+bin/
+tmp/
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.settings/
+.loadpath
+.recommenders
+
+# Eclipse Core
+.project
+
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# PyDev specific (Python IDE for Eclipse)
+*.pydevproject
+
+# CDT-specific (C/C++ Development Tooling)
+.cproject
+
+# JDT-specific (Eclipse Java Development Tools)
+.classpath
+
+# Java annotation processor (APT)
+.factorypath
+
+# PDT-specific (PHP Development Tools)
+.buildpath
+
+# sbteclipse plugin
+.target
+
+# Tern plugin
+.tern-project
+
+# TeXlipse plugin
+.texlipse
+
+# STS (Spring Tool Suite)
+.springBeans
+
+# Code Recommenders
+.recommenders/
+### JDeveloper template
+# default application storage directory used by the IDE Performance Cache feature
+.data/
+
+# used for ADF styles caching
+temp/
+
+# default output directories
+classes/
+deploy/
+javadoc/
+
+# lock file, a part of Oracle Credential Store Framework
+cwallet.sso.lck

--- a/pom.xml
+++ b/pom.xml
@@ -10,11 +10,11 @@
 
   <properties>
     <jena.version>3.0.0</jena.version>
-    <java-rdfa.version>0.4.2-RC2</java-rdfa.version>
+    <java-rdfa.version>0.4.2</java-rdfa.version>
     <slf4j.version>1.6.4</slf4j.version>
     <log4j.version>1.2.16</log4j.version>
     <jruby.version>1.6.5</jruby.version>
-    <poi.version>3.8-beta5</poi.version>
+    <poi.version>3.12</poi.version>
     <junit.version>4.11</junit.version>
     <jdk.version>1.8</jdk.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
@@ -39,11 +39,6 @@
     <dependency>
       <groupId>net.rootdev</groupId>
       <artifactId>java-rdfa</artifactId>
-      <version>${java-rdfa.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>net.rootdev</groupId>
-      <artifactId>java-rdfa-htmlparser</artifactId>
       <version>${java-rdfa.version}</version>
     </dependency>
     

